demo-circuit/src/type_mapping.rs

```diff
@@ -45,12 +45,17 @@ pub type VRFScheme = FieldBasedEcVrf<FieldElement, G2Projective, FieldHash, Grou
 pub type VRFProof = FieldBasedEcVrfProof<FieldElement, G2Projective>;
 pub type VRFPk = G2;
 pub type VRFSk = ScalarFieldElement;
-pub type FieldElementGadget = FpGadget<FieldElement>;
-pub type FieldHashGadget = TweedleFrDensityOptimizedPoseidonHashGadget;
+pub(crate) type FieldElementGadget = FpGadget<FieldElement>;
+pub(crate) type FieldHashGadget = TweedleFrDensityOptimizedPoseidonHashGadget;
+
+// Kept for backward compatibility of Threshold Circuit V1
+pub(crate) type UnoptimizedFieldHashGadget = TweedleFrPoseidonHashGadget;
```

Comment:
Does it make sense to introduce this type? Can't Threshold Circuit V1 directly employ the `TweedleFrPoseidonHashGadget` as in the previous version?

Reply:
I did it in order to make it explicit, so we have less chance to be confused next time D: